### PR TITLE
Homework 5(recursion) done

### PR DESCRIPTION
**Pull request by goharminasyan**

### CODE REVIEW

File: src/algorithms/MathUtil.java [hunk 1]
```diff
@@ -38,7 +41,14 @@ public static int pow(int number, int pow){
...
     public static int factorial(int n) {
-       return 0;
+        int calculatedNumber = n;
+
+        if (n == 1) {
+            return 1;
+        }
+
+        calculatedNumber *= factorial(n - 1);
+        return calculatedNumber;
```
Comment (Karlen999) [inline src/algorithms/MathUtil.java:51]: done

File: src/algorithms/MathUtil.java [hunk 2]
```diff
@@ -49,21 +59,31 @@ public static int factorial(int n) {
...
     public static int abs(int n) {
-        return 0;
-    }
 
+        if (n >= 0) {
+            return n;
+        }
+
+        return abs(n * -1);
+    }
```
Comment (Karlen999) [inline src/algorithms/MathUtil.java:68]: done

File: src/algorithms/MathUtil.java [hunk 3]
```diff
@@ -49,21 +59,31 @@ public static int factorial(int n) {
...
     public static int reverse(int number) {
+        if (number < 10) {
+            reversedNumber = reversedNumber * 10 + number;
+            return number;
+        }
 
-        return 0;
+        reverse(number % 10);
+        reverse(number / 10);
+        return reversedNumber;
```
Comment (Karlen999) [inline src/algorithms/MathUtil.java:86]: done

**Comment by Karlen999** [comment]
reviewed.
good job !!!

**Comment by goharminasyan** [comment]
Thanke you ))

ср, 10 нояб. 2021 г. в 08:37, Karlen ***@***.***>:

**Pull request closed without merging**